### PR TITLE
Feature/connect availability pages

### PR DESCRIPTION
**Pull request by Erbidium**

### CODE REVIEW

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 1]
```diff
@@ -71,6 +73,58 @@ public async Task CreateAvailabilitySlot(NewAvailabilitySlotDto slotDto)
 
             await _context.SaveChangesAsync();
         }
+        
+        public async Task<AvailabilitySlotDto> GetAvailabilitySlotById(long id)
+        {
+            var availabilitySlot = await _context.AvailabilitySlots
+                .Include(slot => slot.AdvancedSlotSettings)
+                .Include(slot => slot.Location)
+                .FirstOrDefaultAsync(_ => _.Id == id);
```
Comment (apofis96) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Don't use underscore as variable name if variable will be used

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 2]
```diff
@@ -71,6 +73,58 @@ public async Task CreateAvailabilitySlot(NewAvailabilitySlotDto slotDto)
...
+                throw new KeyNotFoundException("Availability slot doesn't exist");
+            }
+            return _mapper.Map<AvailabilitySlotDto>(availabilitySlot);
+        }
+
+        public async Task<AvailabilitySlotDto> UpdateAvailabilitySlot(long id, UpdateAvailabilityDto updateAvailabilityDto)
+        {
+            var availabilitySlot = await _context.AvailabilitySlots
+                .Include(slot => slot.AdvancedSlotSettings)
+                .FirstOrDefaultAsync(_ => _.Id == id);
```
Comment (apofis96) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: same

File: backend/EasyMeets.Core/EasyMeets.Core.Common/DTO/Availability/UpdateAvailability/EventDetailsUpdateDto.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿using EasyMeets.Core.Common.Enums;
+
+namespace EasyMeets.Core.Common.DTO.Availability.UpdateAvailability;
+
+public class EventDetailsUpdateDto
+{
+    public bool ZoneChoice { get; set; }
+    public string LinkChoice { get; set; }
```
Comment (apofis96) [inline backend/EasyMeets.Core/EasyMeets.Core.Common/DTO/Availability/UpdateAvailability/EventDetailsUpdateDto.cs]: Fix nullable warnings

**Review by apofis96 — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.Common/DTO/Availability/UpdateAvailability/EventDetailsUpdateDto.cs (hunk 3 above)
Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.Common/DTO/Availability/UpdateAvailability/EventDetailsUpdateDto.cs]: Fixed

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs (hunk 2 above)
Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Fixed

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs (hunk 1 above)
Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Fixed

**Review by apofis96 — APPROVED** [review]

**Pull request merged**